storybook/decorators/withBackgroundMatchingColorScheme.jsx

```diff
@@ -36,5 +36,5 @@ export function withBackgroundMatchingColorScheme(Story, context) {
         }
     }, [colorScheme, context]);
 
-    return <Story />;
+    return story();
```

Comment:
That's an odd one. Is it related to something changed in a newer version of SB?

Reply:
Yep, there were some rendering issues when using `<Story />` instead of `story()`.

If I remember correctly, the error "Rendered more hooks than during the previous render" appeared in React 18 when using the new root API, Strict Mode, and `<Story />`.

Reply:
It's odd since Storybook official docs example still say tp use `<Story />` not `story()`

https://storybook.js.org/docs/react/writing-stories/decorators#global-decorators

Reply:
The documentation mentions both syntaxes, but don't seem to tell when to use one over another 🤔

Reply:
Oh you are right, I haven't notice the tab before you mentionned it.

Well it says the following:

// Replacing the <Story/> element with a Story function is also a good way of writing decorators.
// Useful to prevent the full remount of the component's story.

Preventing a full remount seems like a good idea then.

I wonder thought why it cause rendering issue when using the component syntax. Is there an underlying issue we will hide by patching with the function syntax?